File: src/main/java/com/sfh/pokeRogueBot/model/browser/pokemonjson/Status.java
package com.sfh.pokeRogueBot.model.browser.pokemonjson;

import com.sfh.pokeRogueBot.model.enums.StatusEffect;
import lombok.Data;

@Data
public class Status {

    private StatusEffect effect;
    private int turnCount;

    public float getCatchRateModificatorForStatusEffect(){
        return switch (effect) {
            case NONE -> 1;
            case POISON, PARALYSIS, TOXIC, BURN -> 1.5f;
            case SLEEP, FREEZE -> 2.5f;
            default -> throw new IllegalArgumentException("Unknown status effect: " + effect);
        };
    }
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/browser/pokemonjson/Status.kt
package com.sfh.pokeRogueBot.model.browser.pokemonjson

import com.sfh.pokeRogueBot.model.enums.StatusEffect

data class Status(
    var effect: StatusEffect?,
    var turnCount: Int
) {
    fun getCatchRateModificatorForStatusEffect(): Float {
        return when (effect) {
            StatusEffect.NONE -> 1f
            StatusEffect.POISON, StatusEffect.PARALYSIS, StatusEffect.TOXIC, StatusEffect.BURN -> 1.5f
            StatusEffect.SLEEP, StatusEffect.FREEZE -> 2.5f
            else -> throw IllegalArgumentException("Unknown status effect: $effect")
        }
    }
}